Refactoring

swissspidy opened the pull request:
* More modern syntax
* Use single namespace file instead of god class
* Use GitHub Actions instead of travis
* Introducing linting

swissspidy committed:
First pass at using a simple namespace file

Plugin doesn't really need to be a class on its own, a few functions in the namespace could suffice.

grappler committed:
Update assests

grappler committed:
Update bin/install-wp-tests.sh and travis

grappler committed:
Composer update

Update minimum PHP version to 5.4

grappler committed:
Rename main class Digest to Plugin

grappler committed:
Correct require paths

grappler committed:
Fix missing constants

grappler committed:
Fix tests with missing class

grappler committed:
Remove reference to removed digest function

grappler committed:
Make Registry class methods static

Regester default events outside of the class

grappler committed:
Run PHP Unit with GitHub actions

grappler committed:
Remove travis

grappler commented:
Test Email mit Kommentar und Passwort zurücksetzten.
![image](https://user-images.githubusercontent.com/1785641/102611367-0e0e0400-412f-11eb-89f3-5c84b0d9e38a.png)

ocean90 commented:
Can you set up the [linting workflow](https://github.com/wearerequired/.github/blob/master/workflow-templates/linting.yml) (~without eslint/stylelint~) please?

grappler commented:
> Can you set up the linting workflow (without eslint/stylelint) please?

You mean deactivate eslint/stylelint or create a separate workflow for them?

ocean90 commented:
Ah, I thought the plugin has no CSS/JS files. In this case including the workflow as is would be nice.

grappler committed:
Add linting action

grappler commented:
I have added it but it has not triggered. There are a few other changes that I would like to make like remove grunt but I plan to do that in a separate PR.

ocean90 committed:
Remove unused linting.properties.json file

ocean90 committed:
Lint only PHP for now

ocean90 committed:
Add wearerequired/coding-standards

ocean90 committed:
Update PHPCS config

ocean90 committed:
Update .editorconfig

ocean90 committed:
Rename to .yml

ocean90 commented:
I don't know why it doesn't run. 😞

grappler commented:
It might only run when creating a PR. The other task runs on push.

required-com commented:
`pull_request` should also include the `synchronized` event which is for every new commit on a PR.

ocean90 committed:
Add push

ocean90 committed:
Merge branch 'master' into 14-namespace

ocean90 committed:
Limit push to the default branch

ocean90 commented:
`pull_request` doesn't run if the PR has merge conflicts…

grappler committed:
Update plugable functions for WP 5.6

Ignore PHPCS as code is copied from WordPress

grappler committed:
Fix CS issues

grappler commented:
@ocean90 I have fixed most of the issues. There are only "2 errors and 8 warnings" left. I am not sure what to do with them. Would love to have your feedback.

ocean90 commented on inc/Message/Section.php:
```diff
@@ -66,12 +59,11 @@ abstract class Section implements MessageInterface {
-	 * @param array   $entries The message entries.
-	 * @param WP_User $user    The current user.
+	 * @param array    $entries The message entries.
+	 * @param \WP_User $user The current user.
```
```suggestion
	 * @param \WP_User|null $user The current user.
```

ocean90 commented on inc/Message/CoreUpdate.php:
```diff
@@ -29,13 +26,12 @@ class CoreUpdate extends Section {
-	 * @param WP_User $user    The current user.
-	 * @param string  $event   The current event.
+	 * @param array    $entries The core update entries.
+	 * @param \WP_User $user The current user.
```
```suggestion
	 * @param \WP_User|null $user The current user.
```

ocean90 reviewed (changes requested):
The two errors are due to the incorrect `@param` tag.

For the `current_time()` warnings we'd have to check each usage whether the current time in UTC or in the current timezone needs to be used. If UTC is fine we can switch to `time()`. The others can be ignored with the `phpcs:ignore` comment which should also explain why it's ignored.

grappler committed:
Fix incorrect @param tags

Co-authored-by: Dominik Schilling <dominikschilling+git@gmail.com>

grappler commented:
Thanks @ocean90!

I have changed all of the times to UTC. There maybe an issue for the first emails after upgrading. I am not sure how best to mitigate this.

grappler committed:
Convert all times to UTC

The time is always compared in human_time_diff() thus making no difference

ocean90 commented on inc/Message/CoreUpdate.php:
```diff
@@ -29,13 +26,12 @@ class CoreUpdate extends Section {
-	 * @param array   $entries The core update entries.
-	 * @param WP_User $user    The current user.
-	 * @param string  $event   The current event.
+	 * @param array         $entries The core update entries.
+	 * @param \WP_User|null $user The current user.
+	 * @param string        $event   The current event.
```
```suggestion
	 * @param array         $entries The core update entries.
	 * @param \WP_User|null $user    The current user.
	 * @param string        $event   The current event.
```

ocean90 commented on template-tags/pluggable.php:
```diff
@@ -0,0 +1,177 @@
+		$blogname = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+
+		if ( 'user' !== $notify ) {
+			$switched_locale = switch_to_locale( get_locale() );
```
This function was introduced in WordPress 4.7.0, but the current plugin requirement is 4.4.0. Let's update the "Requires at least" parameter.

ocean90 commented on digest.php:
```diff
@@ -12,8 +10,6 @@
  * License:     GPLv2+
  * Text Domain: digest
  * Domain Path: /languages
```
The updated PHP and WP requirements need to be added to the [plugin headers](https://developer.wordpress.org/plugins/plugin-basics/header-requirements/#header-fields).

ocean90 commented on inc/namespace.php:
```diff
@@ -0,0 +1,277 @@
+ */
+function load_textdomain() {
+	load_plugin_textdomain( 'digest' );
+}
```
```suggestion
```

This can be removed once the WP requirement has been raised to 4.7.

grappler committed:
Set WP & PHP minimum version

- Add test for minimum versions
- Remove text domain code as it is not needed anymore
- Install PHPUnit 5 or 7
- Add mysql extension for PHP 5.6
- Remove Coding Standard when runnuing PHPUnit
- Remove PHP 5.6 incompatable code

ocean90 commented on phpcs.xml.dist:
```diff
@@ -0,0 +1,31 @@
+	<file>.</file>
+
+	<rule ref="Required">
+		<!-- Not compatable with PHP 5.6 -->
```
```suggestion
		<!-- Not compatible with PHP 5.6 -->
```

ocean90 committed:
Update phpcs.xml.dist

grappler committed:
Only run lastest WP on PHP 7.4
